packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts

```diff
@@ -847,6 +870,39 @@ export class GraphqlApi extends GraphqlApiBase {
     return true;
   }
 
+  /**
+   * Add an environment variable to the construct.
+   */
+  public addEnvironmentVariable(key: string, value: string) {
+    if (this.definition.sourceApiOptions) {
+      throw new Error('Environment variables are not supported for merged APIs');
+    }
```

Comment:
Because Environment variables are not supported for merged API. The Cfn errors: 

```
MergedAPI (MergedAPI08D3EAD1) Operation not allowed, Merged API resources are read-only (Service: AmazonDeepdish; Status Code: 400; Error Code: BadRequestException; Request ID: ...
```